### ISSUE TITLE
Avoid spaghetti when reading multiple sheets

### ISSUE THREAD

**Issue by karakasa** [karakasa/PancakeSpreadsheet#5]
As @Intuos5 mentioned [here](https://discourse.mcneel.com/t/pancakespreadsheet-read-write-xls-x-files-without-excel/145611/12?u=gankeyu).

![image](https://user-images.githubusercontent.com/12855165/232006862-29faa6da-5422-4e83-92d3-01c65d2ea995.png)

**Comment by karakasa** [comment]
Due to the complexity of GH's tree processing, I would propose utilizing [Pancake](https://www.food4rhino.com/en/app/pancake)'s Assoc feature.

![image](https://user-images.githubusercontent.com/12855165/232007104-a599b95b-59b4-4fe5-b218-c0e21a06fd31.png)

[read multiple sheets.zip](https://github.com/karakasa/PancakeSpreadsheet/files/11231299/read.multiple.sheets.zip)

In previous releases of PancakeSpreadsheet, the plugin failed to display Assoc-related components if PancakeSpreadsheet was loaded before Pancake. This is fixed in the latest commit.

**Comment by Intuos5** [comment]
@karakasa Didn't know this function yet, this will come in handy with the attributes I set!

I am running into the issue that I name the first cell to the spreadsheet name to be able to see the data in a single panel.
![afbeelding](https://user-images.githubusercontent.com/63149655/232333643-a4e457f7-d3c6-477d-bd48-f078655fa861.png)
![afbeelding](https://user-images.githubusercontent.com/63149655/232338396-874db170-d920-4b39-a896-e56e91f5ffcf.png)
However, this overpopulates the assoc with outputs that have only nulls except for the spreadsheet that contains the data (which is fine, but if I would merge them to a single tree, I have spaghetti again).

**Comment by Intuos5** [comment]
Is there a way for you to sequentially read each spreadsheet in a Read XLS component and construct a list like this:
{Spreadsheet number; Column/ row number}?

---

I do like the Assoc, but in this particular case, I prefer to have the headers of the tables as first item of each branch. That way, I can easily verify the data by looking at the first item for each branch in a panel. If I use the data, I cull the first item in advance.  I think this is worth the additional components, because it makes up for it quickly because it makes for an easier understanding of which number is what.

![afbeelding](https://user-images.githubusercontent.com/63149655/232338048-6980ec3f-3e8d-43fe-b219-041c9e20e593.png)

Another option for me would be to leave the headers in the Assoc, perhaps there could be a boolean input to leave the headers in. Is that something you might be able to add?

I would then have to read the spreadsheet name and add replace the first item in the branch {*;0} to get the result I am looking for. That would also be a good solution for me.

**Comment by Intuos5** [comment]
Just now I thought of another option to parse multiple sheets, which although it does not solve my problem, it may be nice for reading multiple sheets as well. 

The idea is to add a component that outputs each sheet as a tree with each row /colum as branch. Like with the Assoc component, you could rename the outputs of the component by referencing the sheet name.

**Comment by karakasa** [comment]
@Intuos5 I created a new component. Plz try it and see if it satisfies.

![image](https://user-images.githubusercontent.com/12855165/234330612-2fa3cf20-7e99-42a4-8d67-15e44f6ce396.png)
[PancakeSpreadsheet.zip](https://github.com/karakasa/PancakeSpreadsheet/files/11324320/PancakeSpreadsheet.zip)

**Comment by Intuos5** [comment]
@karakasa Awsome! I recently had to bugfix my old script because I went over the 25 sheets I had wired up, but I no longer have to check that now. This works exactly the way I hoped it would. I have replaced the spaghetti components with the new component and the result of the most complex script is still the same. Thank you very much! This is gonna be insanely helpful for my work.
Will you add this to the package manager version of Pancake Spreadsheet too in time?
